clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -492,6 +508,10 @@ static bool isSYCLUndefinedAllowed(const FunctionDecl *Callee,
   if (!Callee->getIdentifier())
     return false;
 
+  if (LibdeviceCmathSet.count(Callee->getName().str()) &&
```

Comment:
One thing that confuses me is: aren't all these functions already builtins? 

https://github.com/intel/llvm/blob/89e2280d53e50346e69ed8354d47a9223f46f483/clang/lib/Sema/SemaSYCL.cpp#L4680

https://github.com/intel/llvm/blob/89e2280d53e50346e69ed8354d47a9223f46f483/clang/include/clang/Basic/Builtins.def#L1350C55-L1350C55

Reply:
Is the issue that the sycl versions of these functions don't get treated as builtins, so we need this new table to identify them?